docs: Add fully-local developer workflow and traps

AndiDog opened the pull request:
I found that only the virtualized toolchain with `make start` is explained, but working locally is a lot faster (webpack rebuilds, Go rebuilds take seconds instead of minutes on macOS x64). Therefore I wrote down a step-by-step guide how to get a whole Kubernetes + Argo CD development environment working locally, including any potential errors and their workarounds.

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [n/a] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [n/a] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [?] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [n/a] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [n/a] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [n/a] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

crenshaw-dev reviewed (changes requested):
This is awesome!

Can you change "ArgoCD" to "Argo CD"? Either way is fine, but in docs we've tended to go with the latter.  :-)

AndiDog commented:
> Can you change "ArgoCD" to "Argo CD"? Either way is fine, but in docs we've tended to go with the latter. :-)

Left it consistent on purpose, but I like that spelling even more. Changed in a new commit, since I replaced it in the whole file.

crenshaw-dev reviewed (approved):
Thanks so much @AndiDog!
